### ISSUE TITLE
Support JS in HTML

### ISSUE THREAD

**Issue by victor-diez-sonarsource** [SonarSource/SonarJS#3597]
https://sonarsource.atlassian.net/jira/software/c/projects/MMF/issues/MMF-2427

**Comment by victor-diez-sonarsource** [comment]
We tested both `eslint-html-parser` and `eslint-plugin-html`.

We managed to make `eslint-plugin-html`, but it required forking it, and the ESLint setup was too specific and using non-exposed methods to override certain behaviours.

`eslint-html-parser` created an AST which was accepted to create a SourceCode object, but then that instance did not seem usable in the Linter.

Instead of depending on already existing ESLint plugins, we chose to follow the same path as the YAML parser, this is using an existing HTML parser to extract the js code ourselves and treat it like standalone js.
